tests/components/signal_messenger/test_notify.py

```diff
@@ -0,0 +1,122 @@
+"""The tests for the signal_messenger platform."""
+
+import os
+import tempfile
+import unittest
+from unittest.mock import patch
+
+from pysignalclirestapi import SignalCliRestApi
+import requests_mock
+
+import homeassistant.components.signal_messenger.notify as signalmessenger
+from homeassistant.setup import async_setup_component
+
+BASE_COMPONENT = "notify"
+
+
+async def test_signal_messenger_init(hass):
+    """Test that service loads successfully."""
+
+    config = {
+        BASE_COMPONENT: {
+            "name": "test",
+            "platform": "signal_messenger",
+            "url": "http://127.0.0.1:8080",
+            "number": "+43443434343",
+            "recipients": ["+435565656565"],
+        }
+    }
+
+    with patch("pysignalclirestapi.SignalCliRestApi.send_message", return_value=None):
+        assert await async_setup_component(hass, BASE_COMPONENT, config)
+        await hass.async_block_till_done()
+
+        # Test that service loads successfully
+        assert hass.services.has_service(BASE_COMPONENT, "test")
+
+
+class TestSignalMesssenger(unittest.TestCase):
+    """Test the signal_messenger notify."""
+
+    def setUp(self):
+        """Set up things to be run when tests are started."""
+        recipients = ["+435565656565"]
+        number = "+43443434343"
+        client = SignalCliRestApi("http://127.0.0.1:8080", number)
+        self._signalmessenger = signalmessenger.SignalNotificationService(
+            recipients, client
+        )
+
+    @requests_mock.Mocker()
+    def test_send_message(self, mock):
+        """Test send message."""
+        message = "Testing Signal Messenger platform :)"
+        mock.register_uri(
```

Comment:
We don't need to test the library requests. We should patch the library and make sure that we are interfacing with the library correctly.

The library is responsible for testing that it makes the correct requests.